src/classes/AFFL_AccRecordType_TDTM.cls

```diff
@@ -146,9 +146,10 @@ public class AFFL_AccRecordType_TDTM extends TDTM_Runnable {
                     if (!containsNew(accsTypeChanged, accNewOldType.oldType)) {
                         contact.put(accNewOldType.oldType, null);
```

Comment:
I'm not exactly sure how this line of code is reached, but I'd expect a NPE to be potentially thrown here as well. How do we enter this logic? Does it get called when switching from a record type that wasn't mapped to a record type that is mapped?

Reply:
@robertbwatson Not sure I am following your question, the updatePrimaryAfflsOnChange method gets called from the run method, if the account already had a record type those would get stored in accTypesChanged, else they would get stored in accTypesAdded so  accNewOldType.oldType wouldn't throw an exception since that is a record from the list accTypesChanged.

Reply:
I'm having a heck of time deciphering what this class as a hole is doing so I have to grok it a bit. Will add my comments as soon as I can. The GH interface by only showing me the changes is making it tough to absorb so I will review the file as a whole and come back and make comments.

Reply:
@skosgi1 Yeah, I was just concerned if accNewOldType.oldType could ever be null because if so, we'd get a NPE here.

It looks like we're checking for null on line 97 while we're building this list (https://github.com/SalesforceFoundation/EDA/blob/fa783e6f7d965aa2b615920950678752868dad76/src/classes/AFFL_AccRecordType_TDTM.cls#L97), so I think this is safe to leave as-is.

Reply:
Ooh, nice - thanks Github for formatting this so nicely :)

Reply:
This could be a candidate for a very specific null check unit test, no?

Reply:
I think this was scenario was captured in changePrimaryAffiliationRecordTypeReentrancyCheck().